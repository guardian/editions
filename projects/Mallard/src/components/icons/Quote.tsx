import React from 'react';
import Svg, { Path } from 'react-native-svg';
import { color } from 'src/theme/color';

const width = 37;
const height = 21;

const Quote = ({ fill = color.text, scale = 1 }) => (
	<Svg
		style={{
			transform: [
				{ scale },
				{ translateY: (height - height * scale) / 2 },
				{ translateX: (width - width * scale) / -2 },
			],
		}}
		width={width}
		height={height}
		fill="none"
	>
		<Path
			fill-rule="evenodd"
			clip-rule="evenodd"
			d="M7.96664 0.75H15.1666C14.3 7.18748 13.5333 13.5 13.2 20.5624H0C1.2 13.6875 3.69999 7.18748 7.96664 0.75ZM24.7332 0.75H31.8332C31.0665 7.18748 30.1999 13.5 29.8665 20.5624H16.6999C18.0666 13.6875 20.4666 7.18748 24.7332 0.75Z"
			fill={fill}
		/>
	</Svg>
);

export { Quote };
