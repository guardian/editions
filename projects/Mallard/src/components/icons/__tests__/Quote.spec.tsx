import React from 'react';
import type { ReactTestRendererJSON } from 'react-test-renderer';
import TestRenderer from 'react-test-renderer';
import { Quote } from '../Quote';

describe('Quote', () => {
	it('should display a Quote icon in SVG', () => {
		const component: ReactTestRendererJSON | null = TestRenderer.create(
			<Quote />,
		).toJSON();
		expect(component).toMatchSnapshot();
	});
});
