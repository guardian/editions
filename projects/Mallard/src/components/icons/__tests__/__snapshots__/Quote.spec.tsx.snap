// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote should display a Quote icon in SVG 1`] = `
<RNSVGSvgView
  bbHeight={21}
  bbWidth={37}
  fill="none"
  focusable={false}
  height={21}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      Object {
        "transform": Array [
          Object {
            "scale": 1,
          },
          Object {
            "translateY": 0,
          },
          Object {
            "translateX": -0,
          },
        ],
      },
      Object {
        "opacity": 1,
      },
      Object {
        "flex": 0,
        "height": 21,
        "width": 37,
      },
    ]
  }
  width={37}
>
  <RNSVGGroup
    fill={null}
    fillOpacity={1}
    fillRule={1}
    font={Object {}}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={
      Array [
        "fill",
      ]
    }
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={null}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth={1}
    vectorEffect={0}
    x={0}
    y={0}
  >
    <RNSVGPath
      d="M7.96664 0.75H15.1666C14.3 7.18748 13.5333 13.5 13.2 20.5624H0C1.2 13.6875 3.69999 7.18748 7.96664 0.75ZM24.7332 0.75H31.8332C31.0665 7.18748 30.1999 13.5 29.8665 20.5624H16.6999C18.0666 13.6875 20.4666 7.18748 24.7332 0.75Z"
      fill={
        Array [
          0,
          4279374354,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      originX={0}
      originY={0}
      propList={
        Array [
          "fill",
        ]
      }
      rotation={0}
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={1}
      vectorEffect={0}
      x={0}
      y={0}
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
